File: server/src/main/java/com/shiros/inventory/entity/Supplier.java
package com.shiros.inventory.entity;

import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;

@Entity
@Table(name="tb_supplier")
@EntityListeners(AuditingEntityListener.class)
public class Supplier {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name="supplier_name")
    private String supplierName;

    @Column(name="supplier_address")
    private String supplierAddress;

    @Column(name="supplier_contact")
    private String supplierContact;

    @Column(name="supplier_contact_name")
    private String supplierContactName;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getSupplierName() {
        return supplierName;
    }

    public void setSupplierName(String supplierName) {
        this.supplierName = supplierName;
    }

    public String getSupplierAddress() {
        return supplierAddress;
    }

    public void setSupplierAddress(String supplierAddress) {
        this.supplierAddress = supplierAddress;
    }

    public String getSupplierContact() {
        return supplierContact;
    }

    public void setSupplierContact(String supplierContact) {
        this.supplierContact = supplierContact;
    }

    public String getSupplierContactName() {
        return supplierContactName;
    }

    public void setSupplierContactName(String supplierContactName) {
        this.supplierContactName = supplierContactName;
    }

}

File: client/src/data/table/tableColumns/supplier.js
import { Link } from 'react-router-dom'
import React from 'react'

export default [
  {
    title: 'Supplier Name',
    dataIndex: 'supplierName',
    key: 'supplierName',
    render: (supplierName, { id }) => { return <Link to={`/supplier/${id}`}>{supplierName}</Link> }
  },
  {
    title: 'Address',
    dataIndex: 'supplierAddress',
    key: 'supplierAddress'
  },
  {
    title: 'Contact',
    dataIndex: 'supplierContact',
    key: 'supplierContact'
  },
  {
    title: 'Contact Name',
    dataIndex: 'supplierContactName',
    key: 'supplierContactName'
  }
]